model-annotator/src/main/java/io/fabric8/kubernetes/ModelRuleFactory.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes;
+
+import org.jsonschema2pojo.rules.Rule;
+import org.jsonschema2pojo.rules.RuleFactory;
+import org.jsonschema2pojo.util.ParcelableHelper;
+
+import com.sun.codemodel.JPackage;
+import com.sun.codemodel.JType;
+
+/**
+ * Model rule factory to support mapping of interfaces.
+ */
+public class ModelRuleFactory extends RuleFactory {
+
+  @Override
+  public Rule<JPackage, JType> getObjectRule() {
+    return new ModelObjectRule(this, new ParcelableHelper(), getReflectionHelper());
+  }
+}
```

Comment:
We already have a [RuleFactory](https://github.com/fabric8io/kubernetes-client/blob/4eee56eb02cc3880435bcf54aacd86c735205013/kubernetes-model-generator/kubernetes-model-jsonschema2pojo/src/main/java/io/fabric8/kubernetes/jsonschema2pojo/Fabric8RuleFactory.java) and a [custom rule](https://github.com/fabric8io/kubernetes-client/blob/b0890fa0535818f066a76c67311b0b6c5a7b6f92/kubernetes-model-generator/kubernetes-model-jsonschema2pojo/src/main/java/io/fabric8/kubernetes/jsonschema2pojo/Fabric8NameHelper.java) for the Name Helper.

Maybe we could add the new rule to the existing Fabric8RuleFactory and avoid the creation of another Factory.
i.e. All Fabric8 jsonschema rule customizations:
- Fabric8RuleFactory
  - Fabric8NameHelper
  - Fabric8ObjectRule

Reply:
+1 will do it

Reply:
updated